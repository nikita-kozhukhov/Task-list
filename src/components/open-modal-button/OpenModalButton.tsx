import { useState } from 'react';
import ReactModal from 'react-modal';

import { ModalCreateTask } from 'components/modal-create-task/ModalCreateTask';
import { Button } from 'ui/button/Button';
import { Modal } from 'ui/modal/Modal';

import styles from './OpenModalButton.module.scss';

ReactModal.setAppElement('#root');

export const OpenModalButton = () => {
  const [isOpen, setIsOpen] = useState(false);

  const handleOpen = () => {
    setIsOpen(true);
  };

  const handleClose = () => {
    setIsOpen(false);
  };

  return (
    <div className={styles['open-modal-button']}>
      <Button label="Создать задачу" variant="primary" onClick={handleOpen} />

      <Modal isOpen={isOpen} onClose={handleClose}>
        <ModalCreateTask onClose={handleClose} />
      </Modal>
    </div>
  );
};
