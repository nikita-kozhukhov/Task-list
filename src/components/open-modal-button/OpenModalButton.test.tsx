import { configureStore } from '@reduxjs/toolkit';
import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import tasksReducer from 'store/slices/tasks';
import { OpenModalButton } from './OpenModalButton';

const store = configureStore({
  reducer: {
    tasks: tasksReducer,
  },
});

describe('OpenModalButton', () => {
  test('рендерит кнопку "Создать задачу"', () => {
    render(<OpenModalButton />);

    expect(
      screen.getByRole('button', { name: /Создать задачу/i })
    ).toBeInTheDocument();
  });

  test('открывает модальное окно при клике на кнопку', () => {
    render(
      <Provider store={store}>
        <OpenModalButton />
      </Provider>
    );

    const openButton = screen.getByRole('button', { name: /Создать задачу/i });
    fireEvent.click(openButton);

    expect(screen.getByText(/Создание задачи/i)).toBeInTheDocument();
  });

  test('закрывает модальное окно при onClose', () => {
    render(
      <Provider store={store}>
        <OpenModalButton />
      </Provider>
    );

    const openButton = screen.getByRole('button', { name: /Создать задачу/i });
    fireEvent.click(openButton);

    const closeButton = screen.getByRole('button', { name: /Закрыть/i });
    fireEvent.click(closeButton);

    expect(screen.queryByText(/Создание задачи/i)).not.toBeInTheDocument();
  });

  test('совпадает с предыдущей версией снапшота', () => {
    const { container } = render(<OpenModalButton />);

    expect(container).toMatchSnapshot();
  });
});
