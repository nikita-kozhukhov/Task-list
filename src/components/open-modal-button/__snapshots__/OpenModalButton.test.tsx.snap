// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`OpenModalButton совпадает с предыдущей версией снапшота 1`] = `
<div>
  <div
    class="open-modal-button"
  >
    <button
      class="button button--primary"
      type="button"
    >
      Создать задачу
    </button>
  </div>
</div>
`;
